File: decryption/decryption.js
Array.prototype.each_slice = function (size, callback){
    for (var i = 0, l = this.length; i < l; i += size){
        callback.call(this, this.slice(i, i + size));
    }
};

function solution() {
    var message_words = "012222 1114142503 0313012513 03141418192102 0113 2419182119021713 06131715070119".split(' '),
        keyed_alphabet = "BHISOECRTMGWYVALUZDNFJKPQX",
        alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ",
        mapped_message = [],
        i = 0;

    for (i; i < message_words.length; i++) {
        var slices = [],
            slice_map = '',
            j = 0;
        message_words[i].split('').each_slice(2, function (slice){
            slices.push(Number(slice.join('')));
        });
        for (j; j < slices.length; j++) {
            slice_map += alphabet[keyed_alphabet.split('').indexOf(alphabet[slices[j]])];
        }
        mapped_message.push(slice_map);
    }
    return mapped_message.join(' ');
}

// node, mocha
module.exports.solution = solution;

console.log(solution());

File: decryption/decryption.rb
# DECRYPTION
# SPONSORING COMPANY:
# CHALLENGE DESCRIPTION:

# For this challenge you are given an encrypted message and a key. You have to determine the encryption and encoding technique and then print out the corresponding plaintext message. You can assume that the plaintext corresponding to this message, and all messages you must handle, will be comprised of only the characters A-Z and spaces; no digits or punctuation.

# INPUT SAMPLE:

# There is no input for this program. The encrypted message and key is:

# message: "012222 1114142503 0313012513 03141418192102 0113 2419182119021713 06131715070119",
# keyed_alphabet: "BHISOECRTMGWYVALUZDNFJKPQX"
# OUTPUT SAMPLE:

# Print out the plaintext message. (in CAPS)
def solution
  encrypted      = '012222 1114142503 0313012513 03141418192102 ' \
                   '0113 2419182119021713 06131715070119'
  keyed_alphabet = 'BHISOECRTMGWYVALUZDNFJKPQX'
  alphabet       = [*'A'..'Z'].join
  encrypted.split(' ').map do |w|
    w.chars.each_slice(2).to_a
     .map { |nn| nn.join.to_i }
     .map { |n| alphabet[keyed_alphabet.index(alphabet[n])] }
     .join
  end.join(' ')
end
puts solution